### PR TITLE
Using stdint.h and stdbool.h for type definitions

### PR DESCRIPTION
**Pull request by akallabeth**
When stdint.h or stdbool.h are detected, use these
standard types for definitions in wtypes.h

### CODE REVIEW

File: include/freerdp/assistance.h [hunk 1]
```diff
@@ -77,7 +77,7 @@ FREERDP_API int freerdp_assistance_decrypt(rdpAssistanceFile* file, const char*
 
 FREERDP_API int freerdp_client_populate_settings_from_assistance_file(rdpAssistanceFile* file, rdpSettings* settings);
 
-FREERDP_API rdpAssistanceFile* freerdp_assistance_file_new();
+FREERDP_API rdpAssistanceFile* freerdp_assistance_file_new(void);
 FREERDP_API void freerdp_assistance_file_free(rdpAssistanceFile* file);
```
Comment (hardening) [inline include/freerdp/assistance.h]: good catch

File: winpr/include/winpr/wtypes.h [hunk 2]
```diff
@@ -49,17 +57,39 @@
 #define NEAR
 #endif
 
+#if defined(HAVE_STDINT_H)
```
Comment (hardening) [inline winpr/include/winpr/wtypes.h]: isn't it not defined(HAVE_STDINT_H) here ?

Comment (akallabeth) [inline winpr/include/winpr/wtypes.h]: @hardening You're right, defining now in main ```CMakeLists.txt``` instead of ```config.h```

**Comment by akallabeth** [comment]
@freerdp-bot Be so kind to test again, iOS does not like us today :/

**Comment by akallabeth** [comment]
@freerdp-bot You seem to test different than me... Can we agree now?

**Comment by akallabeth** [comment]
@freerdp-bot Got ya! Now test properly ;)

**Comment by akallabeth** [comment]
@freerdp-bot test please. What is wrong with you today?

File: winpr/include/winpr/wtypes.h (hunk 2 above)
Comment (hardening) [inline winpr/include/winpr/wtypes.h]: I'm not sure to understand, I was meaning that here it should be
#if not defined(HAVE_STDINT_H)

Comment (akallabeth) [inline winpr/include/winpr/wtypes.h]: #hardening no, this part uses the ```intXX_t``` data types defined in ```stdint.h```

Comment (hardening) [inline winpr/include/winpr/wtypes.h]: @akallabeth ok so I don't understand as what I'm reading is that when stdint.h is detected you (re)define int8_t as char and so on. Am I missing something ?

Comment (akallabeth) [inline winpr/include/winpr/wtypes.h]: @hardening ```typedef <existing> <new>``` is the syntax...

Comment (mfleisz) [inline winpr/include/winpr/wtypes.h]: @akallabeth but the code does #define - not typedef so I think @hardening is right ... there is something wrong here

Comment (akallabeth) [inline winpr/include/winpr/wtypes.h]: Ah, ok, looked at the wrong one.

**Comment by hardening** [comment]
@akallabeth could you rebase on last master ?

**Comment by bmiklautz** [comment]
@akallabeth when we have stdbool shouldn't typedef/define BOOL as boolean?

**Comment by akallabeth** [comment]
@bmiklautz ```stdbool.h``` defines ```bool``` not ```boolean``` and to that ```BOOL``` is already defined (``` winpr/include/winpr/wtypes.h:99```)

**Comment by akallabeth** [comment]
@awakecoding @bmiklautz @hardening Rebased pull, any objections / suggestions?

**Comment by hardening** [comment]
After the last changes, it LGTM +1

File: CMakeLists.txt [hunk 3]
```diff
@@ -339,10 +340,19 @@ else()
...
 endif()
 
+if(HAVE_STDBOOL_H)
+	add_definitions(-DHAVE_STDBOOL_H)
+endif()
+
+if(HAVE_STDINT_H)
+	add_definitions(-DHAVE_STDINT_H)
+endif()
+
```
Comment (bmiklautz) [inline CMakeLists.txt]: What was the reason that you've added this as parameter but not in config.h(.in)?

File: config.h.in [hunk 4]
```diff
@@ -4,7 +4,6 @@
 /* Include files */
 #cmakedefine HAVE_FCNTL_H
 #cmakedefine HAVE_UNISTD_H
-#cmakedefine HAVE_STDINT_H
```
Comment (bmiklautz) [inline config.h.in]: What is the reason for removing this?

File: libfreerdp/common/assistance.c [hunk 5]
```diff
@@ -1193,7 +1193,7 @@ int freerdp_client_populate_settings_from_assistance_file(rdpAssistanceFile* fil
 	return 1;
 }
 
-rdpAssistanceFile* freerdp_assistance_file_new()
+rdpAssistanceFile* freerdp_assistance_file_new(void)
```
Comment (bmiklautz) [inline libfreerdp/common/assistance.c]: nice!

File: CMakeLists.txt (hunk 3 above)
Comment (akallabeth) [inline CMakeLists.txt]: @bmiklautz didn't want to include ```config.h``` in ```wtypes.h``` which would trigger a complete rebuild after every change in ```config.h```

File: config.h.in (hunk 4 above)
Comment (akallabeth) [inline config.h.in]: @bmiklautz Duplicate definition as it is already defined via command line

File: CMakeLists.txt (hunk 3 above)
Comment (hardening) [inline CMakeLists.txt]: isn't wtypes.h a public header ? So that means that third party app must define HAVE_STDBOOL_H and HAVE_STDINT_H ?

Comment (akallabeth) [inline CMakeLists.txt]: @hardening True, have not thought about that :/ Seems to be more challenging than expected.

**Comment by bmiklautz** [comment]
Nice idea @akallabeth. I like it very much that with your latest change 
With the latest change the installed version always reflects the compile time settings for types as well. Which we hadn't before \o/. I'll do some testing with this PR.

**Comment by bmiklautz** [comment]
+1
Nice!

**Pull request merged**